pageserver/src/bin/pageserver_binutils.rs

```diff
@@ -110,10 +112,23 @@ fn handle_metadata(path: &Path, arg_matches: &clap::ArgMatches) -> Result<(), an
         );
         update_meta = true;
     }
+    if let Some(latest_gc_cuttoff) = arg_matches.get_one::<String>("latest_gc_cuttoff") {
+        meta = TimelineMetadata::new(
+            meta.disk_consistent_lsn(),
+            meta.prev_record_lsn(),
+            meta.ancestor_timeline(),
+            meta.ancestor_lsn(),
+            Lsn::from_str(latest_gc_cuttoff)?,
+            meta.initdb_lsn(),
+            meta.pg_version(),
+        );
+        update_meta = true;
+    }
 
     if update_meta {
         let metadata_bytes = meta.to_bytes()?;
-        std::fs::write(path, metadata_bytes)?;
+        let mut file = fs::OpenOptions::new().write(true).open(path)?;
```

Comment:
Why do we need it?
`std::fs::write` truncates file if it exists, so there is a risk of loosing metadata file if system is crashed while prerformig this operation.  But to avoid it we can first write metadata to a temporary file and then atomically rename it. `fs::OpenOptions::new().write(true)` doesn't truncate file but overwrite it.
If for some reasons old metadata file is larger then new one (right now it is not possible, but assume that we remove soe fields from it), then there will be garbage at the end of file.
So  I am not sure that it is correct change.

Reply:
reverted